packages/fiori-elements-writer/src/index.ts

```diff
@@ -223,6 +233,19 @@ async function generate<T extends {}>(basePath: string, data: FioriElementsApp<T
         await applyCAPUpdates(fs, feApp.service.capService, settings);
     }
 
+    if (
+        feApp.appOptions?.addAnnotations &&
+        TemplateTypeAttributes[feApp.template.type]?.annotationGenerationSupport?.[feApp.service.version]
+    ) {
+        await writeAnnotations(basePath, feApp, fs);
+    } else {
+        log?.warn(
```

Comment:
This log will output all the time even when the app option is not provided. Can you leave the non-app option conditions here and move the rest back to the `writeAnnotations` function please:
```
if (feApp.appOptions?.addAnnotations) {
    await writeAnnotations(basePath, feApp, fs);
}
```

The context of this file should be to deal with top level options, delegating the detailed implementation to the called functions.